### PR TITLE
Big Sky entry point

### PR DESCRIPTION
**Pull request by dereksmart**
Entry point for Big Sky in the onboarding flow in staging environment. 

- [ ] Image/animation for the card (to do in follow-up)
- [ ] Finalize copy (to do in follow-up)
- [x] Remove the step for inputting the prompt. Just shows loading page. 
- [x] Hide on narrow viewports
- [x] Hide on certain plans 
- [x] Only show with certain goals
- [x] Check if homepage exists before creating one
- [x] Handle browser navigation: If navigating "back" to the loading page, it should actually go to the previous step of design-choices. 

## Proposed Changes

Entry point for Big Sky in the onboarding flow. 

It adds a new card to the /design-choices step, if the conditions of the new `useIsBigSkyEligible()` are met, which are: 
- On a premium (explorer) plan
- Have selected specific goals of `promote` or `other`. 
- is the config enabled (not yet in production)
- Is the viewport wide enough (temporary workaround to UX issues in Big Sky on narrow views)

Follow-ups will be deployed for: 
- Finalized copy for the design choice 
- Finalized image for the design choice card 

## Testing Instructions

Purchase an explorer plan
- can start at http://calypso.localhost:3000/start/premium/
- Choose a domain
- Purchase the plan (explorer)
- You should see the Big Sky card in design-choices, click it
- You should see a loading page, which then takes you to the site editor
- Big Sky should appear

Also test: 
- Go "back" after landing in Big Sky. It should take you back to design-choices
- It should not show on any other plans (only explorer) 

You can also go straight to `http://calypso.localhost:3000/setup/site-setup/design-choices?siteSlug=SITESLUG` to target a specific test site. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx [hunk 1]
```diff
@@ -38,7 +42,7 @@ const DesignChoicesStep: Step = ( { navigation, flow, stepName } ) => {
 			destination: kebabCase( destination ),
 		} );
 
-		if ( destination === 'pattern-assembler' ) {
+		if ( destination === 'pattern-assembler' || destination === 'loadingBigSky' ) {
```
Comment (dereksmart) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx]: haven't tested if this actually works to set the assembler theme.

File: packages/design-picker/src/components/assets/images/hi-big-sky.png
Comment (benzhu56) [inline packages/design-picker/src/components/assets/images/hi-big-sky.png:1]: I think we want a png with better quality. The line/arc aren't smooth and is a bit jarring to look imo.

**Review by benzhu56 — APPROVED** [review]
Tested happy path that big sky loads correctly. Also tested that free plan won't load it. 
Code LG. 👍 
One minor pick for the image quality:

**Review by kthai15 — APPROVED** [review]
Tests well for me

- Tested on a my site on the explorer plan, made it through the happy path.
- Tested on my free site, I don't see the Big Sky card. Tried hitting the Big Sky link directly, which redirected me
- Tested on a site I don't own, didn't see the Big Sky card. Tried hitting the Big Sky link directly, which redirected me

**Pull request merged**